gulpfile.mjs

```diff
@@ -2192,13 +2191,20 @@ gulp.task(
       console.log();
       console.log("### Cloning baseline distribution");
 
-      rimraf.sync(DIST_DIR);
+      fs.rmSync(DIST_DIR, { recursive: true, force: true });
       fs.mkdirSync(DIST_DIR, { recursive: true });
       safeSpawnSync("git", ["clone", "--depth", "1", DIST_REPO_URL, DIST_DIR]);
 
       console.log();
       console.log("### Overwriting all files");
-      rimraf.sync(path.join(DIST_DIR, "*"));
+
+      // Remove all files/folders, except for `.git` because it needs to be a
+      // valid Git repository for the Git commands in the `dist` target to work.
+      for (const entry of fs.readdirSync(DIST_DIR)) {
+        if (entry !== ".git") {
```

Comment:
This change is what fixes the regression from #15975. The initial attempt to upgrade `rimraf` caused the `.git` directory to also be removed because the behavior for wildcards had changed. This was already found in https://github.com/mozilla/pdf.js/issues/15975#issuecomment-1446000986 but I have also verified that locally now by inspecting the state of the directory after `rimraf.sync(path.join(DIST_DIR, "*"));` ran (indeed, everything was gone _except_ for `.git`). This code is fully equivalent and also results in an empty directory where only the `.git` directory is kept. This is an important detail because otherwise the Git commands later on will fail because it's not a valid Git repository, which is what happened last time. Since this also really wasn't obvious at first to me when reading the code, I have included a comment to indicate this.